lib/class-wp-rest-object-search-handler.php

```diff
@@ -0,0 +1,96 @@
+<?php
+/**
+ * REST API: WP_REST_Object_Search_Handler class
+ *
+ * @package gutenberg
+ * @since 3.3.0
+ */
+
+/**
+ * Core base class representing a search handler for an object type in the REST API.
+ *
+ * @since 3.3.0
+ */
+abstract class WP_REST_Object_Search_Handler {
```

Comment:
To match the naming scheme of other abstract classes in the REST API `WP_REST_Meta_Fields`, `WP_REST_Controller`, the class name should just be `WP_REST_Search_Handler`, the implementing classes would follow the naming scheme you uses for `WP_REST_Post_Search_Handler`.

Reply:
Makes sense to me!